src/handlers/payout/post.ts

```diff
@@ -243,28 +244,63 @@ const generatePermitForComments = async (
  * @param incentives - The basic price table for reward calculation
  * @returns - The reward value
  */
-const calculateRewardValue = (comments: Record<string, string[]>, incentives: Incentives): Decimal => {
+const calculateRewardValue = (comments: Record<MarkdownItem, string[]>, incentives: Incentives): Decimal => {
   let sum = new Decimal(0);
-  for (const key of Object.keys(comments)) {
-    const value = comments[key];
+  for (const item of MarkdownItems) {
+    const value = comments[item];
 
     // if it's a text node calculate word count and multiply with the reward value
-    if (key == "#text") {
+    if (item === MarkdownItem.Text) {
       if (!incentives.comment.totals.word) {
         continue;
       }
       const wordReward = new Decimal(incentives.comment.totals.word);
       const reward = wordReward.mul(value.map((str) => str.trim().split(" ").length).reduce((totalWords, wordCount) => totalWords + wordCount, 0));
       sum = sum.add(reward);
     } else {
-      if (!incentives.comment.elements[key]) {
+      const htmlTag = MarkdownItemToHTMLTag[item];
+      if (!htmlTag || !incentives.comment.elements[htmlTag]) {
         continue;
       }
-      const rewardValue = new Decimal(incentives.comment.elements[key]);
+      const rewardValue = new Decimal(incentives.comment.elements[htmlTag]);
       const reward = rewardValue.mul(value.length);
       sum = sum.add(reward);
     }
   }
 
   return sum;
 };
+
+const MarkdownItemToHTMLTag: Record<MarkdownItem, string> = {
+  [MarkdownItem.Text]: HTMLItem.P,
+  [MarkdownItem.Paragraph]: HTMLItem.P,
+  [MarkdownItem.Heading]: HTMLItem.H1,
+  [MarkdownItem.Heading1]: HTMLItem.H1,
+  [MarkdownItem.Heading2]: HTMLItem.H2,
+  [MarkdownItem.Heading3]: HTMLItem.H3,
+  [MarkdownItem.Heading4]: HTMLItem.H4,
+  [MarkdownItem.Heading5]: HTMLItem.H5,
+  [MarkdownItem.Heading6]: HTMLItem.H6,
+  [MarkdownItem.ListItem]: HTMLItem.LI,
+  [MarkdownItem.List]: HTMLItem.UL,
+  [MarkdownItem.Link]: HTMLItem.A,
+  [MarkdownItem.Image]: HTMLItem.IMG,
+  [MarkdownItem.BlockQuote]: HTMLItem.BLOCKQUOTE,
+  [MarkdownItem.Code]: HTMLItem.PRE,
+  [MarkdownItem.Emphasis]: HTMLItem.EM,
+  [MarkdownItem.Strong]: HTMLItem.STRONG,
+  [MarkdownItem.Delete]: HTMLItem.DEL,
+  [MarkdownItem.HTML]: HTMLItem.HTML,
+  [MarkdownItem.InlineCode]: HTMLItem.CODE,
+  [MarkdownItem.LinkReference]: HTMLItem.A,
+  [MarkdownItem.ImageReference]: HTMLItem.IMG,
+  [MarkdownItem.FootnoteReference]: HTMLItem.SUP,
+  [MarkdownItem.FootnoteDefinition]: HTMLItem.P,
```

Comment:
I'm not sure how can we represent footnote reference and definition by HTML? @pavlovcik 

Example of footnote[^1]

[^1]: Footnote 1

Reply:
Very cool. I didn't know about this but I guess we can just file an issue and kick the can down the road?